src/Symfony/Component/BrowserKit/HttpBrowser.php

```diff
@@ -82,7 +82,11 @@ private function getBodyAndExtraHeaders(Request $request, array $headers): array
         $fields = $request->getParameters();
 
         if ($uploadedFiles = $this->getUploadedFiles($request->getFiles())) {
-            $part = new FormDataPart(array_merge($fields, $uploadedFiles));
+            if (\count($fields) !== \count($fields, \COUNT_RECURSIVE)) {
+                $part = new FormDataPart(array_merge_recursive($fields, $uploadedFiles));
```

Comment:
be careful about `array_merge_recursive`. It can do weird things in case of associative arrays. See https://friendsofsymfony.github.io/slides/config_definition.html#6 (and the next slide showing the result). Slides were in French, but the code snippets should still be understandable without the comments anyway.

Reply:
@stof alright, I changed it to `array_replace_recursive` instead